modules/presence/hash.c

```diff
@@ -793,4 +793,48 @@ int delete_cluster_query(str* pres_uri, int event, unsigned int hash_code)
 	return 0;
 }
 
+/* 
+ * entry must be locked before calling this function 
+ * Returns 1 if 
+            - entry exisits subscription is expired. 
+			- entry does not exist.
+ * Returns 0 if entry is found and subscription is active. 
+*/
+int is_sub_expired(shtable_t htable,unsigned int hash_code,str to_tag)
+{
+	subs_t* s= NULL;
+	int is_expired = 0, found = 0;
+
+	s = htable[hash_code].entries->next;
+
+	while(s)
+	{
+		if(s->to_tag.len== to_tag.len &&
+				strncmp(s->to_tag.s, to_tag.s, to_tag.len)== 0)
+		{	
+			found = 1;
+			if(s->expires == 0 && 
+					s->status == TERMINATED_STATUS &&
+						strcasecmp(s->reason.s, "timeout") == 0)
+			{
+				LM_WARN("sub found and expired to_tag [%.*s]\n",to_tag.len, to_tag.s);
```

Comment:
Those messages are good for debugging purpose, is it too much for live system?

Reply:
Maybe `LM_INFO` is a good logging level for those messages?

We want to be able to check if we are hitting this condition but don't the overhead of `LM_DEBUG`

Reply:
Changed log level to INFO